src/pages/AuthenticationPage/AuthenticationComp/AuthSignUp/AuthSignUp.jsx

```diff
@@ -0,0 +1,175 @@
+import { useLocation, useNavigate } from "react-router-dom";
+import "../AuthenticationComp.css";
+import { useState } from "react";
+import {
+  FaUserAlt,
+  MdEmail,
+  BsKeyFill,
+  BsFillEyeSlashFill,
+  BsFillEyeFill,
+} from "../../../../icons/icons";
+import { Loading, Navbar } from "../../../../components/components";
+import { useInputHandler } from "../../../../hooks/customHooks";
+import { useAuthContext } from "../../../../context/AuthContext";
+
+function AuthSignUp() {
+  const [passwordShown, setPasswordShown] = useState(false);
+  const [formSubmitState, setformSubmitState] = useState(false);
+  const { signUpUser } = useAuthContext();
+  function toggleShowPassword() {
+    setPasswordShown((passwordShown) => !passwordShown);
+  }
+  const navigate = useNavigate();
+  const { inputState, inputUpdate } = useInputHandler({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+    password2: "",
+  });
+// passwordValidation is not being used currently but will be used in future.
+  const passwordValidation = () => {
+    const passwordExpression =
+    /^(?=.*\d)(?=.*[a-z]).{5,10}$/;
+
+    if (inputState.password !== inputState.password2) {
+      console.log("passwords do not match");
+      return false;
+    }
+    if (!passwordExpression.test(inputState.password) || !inputState.password2.match(passwordExpression)){
+      console.log(!passwordExpression.test(inputState.password),'1');
+      console.log((inputState.password2.match(passwordExpression)),"2");
+      console.log("password should follow rules");
+      return false
+    }
+    return true;
+  };
+
+  
+
+  function submitFormHandler(e) {
+    e.preventDefault();
+      signUpUser(inputState.firstName,inputState.lastName,inputState.email,inputState.password,setformSubmitState);;
```

Comment:
Instead of passing everything as a separate argument to the function. You can pass only 2 argument: `inputState` object  and `setformSubmitState`. And destructure `inputState` in your `signUpUser` function.

Reply:
Made the changes.